src/main/java/org/alliancegenome/curation_api/services/validation/dto/Gff3DtoValidator.java

```diff
@@ -0,0 +1,177 @@
+		Map<String, String> attributes = getAttributes(dto, dataProvider);
+		String uniqueId = Gff3UniqueIdHelper.getExonOrCodingSequenceUniqueId(dto, attributes, dataProvider);
+		SearchResponse<CodingSequence> searchResponse = codingSequenceDAO.findByField("uniqueId", uniqueId);
+		if (searchResponse != null && searchResponse.getSingleResult() != null) {
+			cds = searchResponse.getSingleResult();
+		} else {
+			cds = new CodingSequence();
+			cds.setUniqueId(uniqueId);
+		}
+		
+		if (attributes.containsKey("Name")) {
+			cds.setName(attributes.get("Name"));
+		}
+		
+		ObjectResponse<CodingSequence> cdsResponse = validateGffEntity(cds, dto, attributes, dataProvider);
+	
+		if (cdsResponse.hasErrors()) {
+			throw new ObjectValidationException(dto, cdsResponse.errorMessagesString());
+		}
+		
+		return codingSequenceDAO.persist(cdsResponse.getEntity());
+	}
+	
+	@Transactional
+	public Transcript validateTranscriptEntry(Gff3DTO dto, BackendBulkDataProvider dataProvider) throws ObjectValidationException {
+
+		Transcript transcript = null;
+		
+		Map<String, String> attributes = getAttributes(dto, dataProvider);
+		if (attributes.containsKey("ID")) {
+			SearchResponse<Transcript> searchResponse = transcriptDAO.findByField("modInternalId", attributes.get("ID"));
+			if (searchResponse != null && searchResponse.getSingleResult() != null) {
+				transcript = searchResponse.getSingleResult();
+			} else {
+				transcript = new Transcript();
+				transcript.setModInternalId(attributes.get("ID"));
+			}	
+		}
+		
+		SearchResponse<SOTerm> soResponse = soTermDAO.findByField("name", dto.getType());
+		if (soResponse != null) {
+			transcript.setTranscriptType(soResponse.getSingleResult());
+		}
+		
+		if (attributes.containsKey("Name")) {
+			transcript.setName(attributes.get("Name"));
+		}
+		
+		ObjectResponse<Transcript> transcriptResponse = validateGffEntity(transcript, dto, attributes, dataProvider);
+		if (!attributes.containsKey("ID")) {
+			transcriptResponse.addErrorMessage("attributes - ID", ValidationConstants.REQUIRED_MESSAGE);
+		}
+		
+		if (transcriptResponse.hasErrors()) {
+			throw new ObjectValidationException(dto, transcriptResponse.errorMessagesString());
+		}
+		
+		return transcriptDAO.persist(transcriptResponse.getEntity());
+	}
+	
+	private <E extends GenomicEntity> ObjectResponse<E> validateGffEntity(E entity, Gff3DTO dto, Map<String, String> attributes, BackendBulkDataProvider dataProvider) {
+		ObjectResponse<E> geResponse = new ObjectResponse<E>();
+		
+		if (attributes.containsKey("curie")) {
+			entity.setModEntityId(attributes.get("curie"));
+		}
+		
+		entity.setDataProvider(dataProviderService.createOrganizationDataProvider(dataProvider.sourceOrganization));
+		entity.setTaxon(ncbiTaxonTermService.getByCurie(dataProvider.canonicalTaxonCurie).getEntity());
+		
+		geResponse.setEntity(entity);
+		
+		return geResponse;
+	}
+	
+	private Map<String, String> getAttributes (Gff3DTO dto, BackendBulkDataProvider dataProvider) {
```

Comment:
🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.MethodParamPadCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'(' is preceded with whitespace.